File: ESTRUTURAS/1162/main.rb
def least_swaps(l, carriages)
  swaps = 0
  (0...l - 1).each do |i|
    (i + 1...l).each do |j|
      swaps += 1 if carriages[i] > carriages[j]
    end
  end
  swaps
end

n = gets.to_i
n.times do
  l = gets.to_i
  carriages = gets.split.map(&:to_i)
  puts "Optimal train swapping takes #{least_swaps(l, carriages)} swaps."
end

File: ESTRUTURAS/1162/main.c
#include <stdio.h>

int leastSwaps(int l, int carriages[l]) {
    int swaps = 0;
    for (int i = 0; i < l - 1; i++)
        for (int j = i + 1; j < l; j++)
            swaps += carriages[i] > carriages[j];
    return swaps;
}

int main() {
    int n, l;
    for (scanf("%d %d", &n, &l); n--; scanf("%d", &l)) {
        int carriages[l];
        for (int i = 0; i < l; i++) scanf("%d", &carriages[i]);
        printf("Optimal train swapping takes %d swaps.\n", leastSwaps(l, carriages));
    }
    return 0;
}
